extensions/mapstore/src/main/java/com/hazelcast/mapstore/GenericMapLoader.java

```diff
@@ -0,0 +1,366 @@
+        sql.execute(
+                "CREATE MAPPING \"" + mappingName + "\""
+                + " EXTERNAL NAME \"" + tableName + "\" "
+                + " TYPE " + deriveMappingType()
+                + " OPTIONS ("
+                + "    '" + OPTION_EXTERNAL_DATASTORE_REF + "' = '" + externalDataStoreRef + "' "
+                + ")"
+        ).close();
+    }
+
+    private SqlRowMetadata loadMetadataFromMapping(String mapping) {
+        try (SqlResult result = sql.execute("SELECT * FROM \"" + mapping + "\" LIMIT 0")) {
+            return result.getRowMetadata();
+        }
+    }
+
+    private void readExistingMapping() {
+        logger.fine("Reading existing mapping for map" + mapName);
+        try (SqlResult sqlResult = sql.execute("SHOW MAPPINGS")) {
+            for (SqlRow sqlRow : sqlResult) {
+                String name = sqlRow.getObject(MAPPING_NAME_COLUMN);
+                if (name.equals(this.mapping)) {
+                    SqlRowMetadata rowMetadata = loadMetadataFromMapping(name);
+                    validateColumns(rowMetadata);
+                    columnMetadataList = rowMetadata.getColumns();
+                    queries = new Queries(name, properties.idColumn, columnMetadataList);
+                    break;
+                }
+            }
+        } catch (Exception e) {
+            initFailure = e;
+        }
+    }
+
+    private void validateColumns(SqlRowMetadata rowMetadata) {
+        Stream.concat(of(properties.idColumn), properties.columns.stream())
+                .distinct() // avoid duplicate id column if present in columns property
+                .forEach(columnName -> validateColumn(rowMetadata.findColumn(columnName), columnName));
+    }
+
+    private int validateColumn(int column, String columnName) {
+        if (column == COLUMN_NOT_FOUND) {
+            throw new HazelcastException("Column '" + columnName + "' not found");
+        }
+        return column;
+    }
+
+    @Override
+    public void destroy() {
+        awaitInitFinished();
+        dropMapping(mapping);
+    }
+
+
+    @Override
+    public GenericRecord load(K key) {
+        awaitInitFinished();
+
+        try (SqlResult queryResult = sql.execute(queries.load(), key)) {
+            Iterator<SqlRow> it = queryResult.iterator();
+            if (it.hasNext()) {
+                SqlRow row = it.next();
+                if (it.hasNext()) {
+                    throw new IllegalStateException("multiple matching rows for a key " + key);
+                }
+                return sqlRowConverter.convertRowToGenericRecord(row, properties);
+            } else {
+                return null;
+            }
+        }
+    }
+
+    /**
+     * Size of the {@code keys} collection is limited by {@link ClusterProperty#MAP_LOAD_CHUNK_SIZE}
+     */
+    @Override
+    public Map<K, GenericRecord> loadAll(Collection<K> keys) {
+
+        awaitInitFinished();
```

Comment:
super nitty nit: you have blank lines on the beginning of the methods and on the end, please remove them (IIRC it will be more in line with Java code guidelines)